<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>558</width>
    <height>694</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::ActionsContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Qt CAN Demo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Receive</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QTableWidget" name="tableWidgetOutput">
          <property name="font">
           <font>
            <family>System</family>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="gridStyle">
           <enum>Qt::DotLine</enum>
          </property>
          <property name="sortingEnabled">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
          <property name="cornerButtonEnabled">
           <bool>false</bool>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>11</number>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>40</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderDefaultSectionSize">
           <number>12</number>
          </attribute>
          <attribute name="verticalHeaderHighlightSections">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderMinimumSectionSize">
           <number>10</number>
          </attribute>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Transmit</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Message</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>ID</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxID">
               <property name="maximum">
                <number>536870911</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="checkBoxExtID">
               <property name="text">
                <string>Extended</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_6">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>DLC</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxDLC">
               <property name="maximum">
                <number>8</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_5">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Data 0..7</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData0">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData1">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData2">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData3">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData4">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData5">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData6">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxData7">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Channel</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QComboBox" name="comboBoxChannel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>200</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonRefresh">
               <property name="maximumSize">
                <size>
                 <width>228</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Refresh</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxExclusive">
             <property name="text">
              <string>Exclusive</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxSilentMode">
             <property name="text">
              <string>Silent mode</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Bus Parameters</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Bitrate:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBoxBusSpeed"/>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>bit/s</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Receive</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QCheckBox" name="checkBoxHexId">
             <property name="text">
              <string>ID in hex</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxHexFlags">
             <property name="text">
              <string>Flags in hex</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxHexData">
             <property name="text">
              <string>Data in hex</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionBusOn"/>
   <addaction name="actionBusOff"/>
   <addaction name="actionTransmit"/>
  </widget>
  <action name="actionBusOn">
   <property name="icon">
    <iconset resource="graphics.qrc">
     <normaloff>:/new/icons/play64.png</normaloff>:/new/icons/play64.png</iconset>
   </property>
   <property name="text">
    <string>Bus on</string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
  </action>
  <action name="actionBusOff">
   <property name="icon">
    <iconset resource="graphics.qrc">
     <normaloff>:/new/icons/stop64.png</normaloff>:/new/icons/stop64.png</iconset>
   </property>
   <property name="text">
    <string>Bus off</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="actionTransmit">
   <property name="icon">
    <iconset resource="graphics.qrc">
     <normaloff>:/new/icons/send64.png</normaloff>:/new/icons/send64.png</iconset>
   </property>
   <property name="text">
    <string>Transmit</string>
   </property>
   <property name="toolTip">
    <string>Transmit</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="graphics.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButtonRefresh</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshChannels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>currentChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshChannels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxChannel</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBusOn</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>goBusOn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBusOff</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>goBusOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxBusSpeed</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setBitrate(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTransmit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>268</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshChannels()</slot>
  <slot>setChannel(int)</slot>
  <slot>goBusOn()</slot>
  <slot>goBusOff()</slot>
  <slot>setBitrate(int)</slot>
  <slot>updateMaxRows(int)</slot>
  <slot>sendMessage()</slot>
 </slots>
</ui>
